webpack/assets/javascripts/react_app/components/Layout/Layout.js

```diff
@@ -0,0 +1,155 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+
+import { VerticalNav } from 'patternfly-react';
+import { isEmpty } from 'lodash';
+import { noop } from '../../common/helpers';
+
+import TaxonomySwitcher from './components/TaxonomySwitcher';
+import UserDropdowns from './components/UserDropdowns';
+import './layout.scss';
+
+class Layout extends React.Component {
+  componentDidMount() {
+    const {
+      items,
+      data,
+      fetchMenuItems,
+      changeLocation,
+      currentLocation,
+      changeOrganization,
+      currentOrganization,
+    } = this.props;
+    if (items.length === 0) fetchMenuItems(data);
+
+    if (!isEmpty(data.locations.current_location)
+     && currentLocation !== data.locations.current_location) {
+      changeLocation(data.locations.current_location);
+    }
+
+    if (!isEmpty(data.organizations.current_org)
+     && currentOrganization !== data.organizations.current_org) {
+      changeOrganization(data.organizations.current_org);
+    }
+  }
+
+  render() {
+    const {
+      items,
+      data,
+      isLoading,
+      changeActiveMenu,
+      changeOrganization,
+      changeLocation,
+      currentOrganization,
+      currentLocation,
+    } = this.props;
+    return (
+      <VerticalNav
+        hoverDelay={0}
+        items={items}
+        onItemClick={changeActiveMenu}
+        {...this.props}
+      >
+        <VerticalNav.Masthead>
+          <VerticalNav.Brand
+            title="foreman"
+            iconImg={data.logo}
+            href={data.root}
+          />
+          <TaxonomySwitcher
+            taxonomiesBool={data.taxonomies}
+            currentLocation={currentLocation}
+            locations={data.locations.available_locations}
+            onLocationClick={changeLocation}
+            currentOrganization={currentOrganization}
+            organizations={data.organizations.available_organizations}
+            onOrgClick={changeOrganization}
+            isLoading={isLoading}
+          />
+          <UserDropdowns
```

Comment:
should we move the drawer here too? AFAIR its forced into this position by using CSS?

Reply:
The drawer is mounted inside `UserDropdowns`
https://github.com/theforeman/foreman/pull/5898/files#diff-228f28407146c3681fe506f35f80de9bR22